### PR TITLE
Add deployment via Kamal

### PR DESCRIPTION
**Pull request by parndt**
This complements the deployment added to marine-defrag/marine-defrag-server#112

We first build the application inside a temporary container running the
specified node version, and then we deploy its files to the production
container so that it can be deployed to the server.

I've had to patch `WebpackGitHash.prototype.getSkipHash` to support the
commit SHA being passed in, as Docker will not be able to access the
repository during build.

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  workflow_dispatch:
+
+env:
+  IMAGE_NAME: ${{ github.repository }}
+  REGISTRY: ghcr.io
+  SERVICE: marine-defrag-client
+
+jobs:
+  build-and-push-image:
+    if: ${{ github.ref == 'refs/heads/master' || github.ref == 'refs/heads/deployment' }}
```
Comment (parndt) [inline .github/workflows/deploy.yml:19]:
```suggestion
    if: ${{ github.ref == 'refs/heads/master' }}
```

before we merge

**Comment by tmfrnz** [comment]
@parndt please note that I had to remove a few lines in webpack as it was resulting in build errors (guess that merge for this PR was a bit premature =) )
https://github.com/marine-defrag/marine-defrag-client/commit/1245f3a8d222a821e76578df616e0d9ac6843d10

**Comment by parndt** [comment]
@tmfrnz all good - can you think of another way to pass that info through?

**Comment by parndt** [comment]
I wonder if we could just rely on there being, or not being, an environment variable before we even define the function:

```js
if (process.env.COMMIT_SHA) {
  WebpackGitHash.prototype.getSkipHash = function(length) {
    return process.env.COMMIT_SHA.substr(0, 7);
  }
}
```

**Pull request merged**